packages/contracts-bedrock/src/L1/OptimismPortal.sol

```diff
@@ -361,6 +430,53 @@ contract OptimismPortal is Initializable, ResourceMetering, ISemver {
         }
     }
 
+    /// @notice Entrypoint to depositing an ERC20 token as a custom gas token.
+    /// @param _to         Target address on L2.
+    /// @param _mint       Units of ERC20 token to deposit into L2.
+    /// @param _value      Units of ERC20 token to send on L2 to the recipient.
+    /// @param _gasLimit   Amount of L2 gas to purchase by burning gas on L1.
+    /// @param _isCreation Whether or not the transaction is a contract creation.
+    /// @param _data       Data to trigger the recipient with.
+    function depositERC20Transaction(
+        address _to,
+        uint256 _mint,
+        uint256 _value,
+        uint64 _gasLimit,
+        bool _isCreation,
+        bytes memory _data
+    )
+        public
+        metered(_gasLimit)
+    {
+        // Can only be called if an ERC20 token is used for gas paying on L2
+        (address token, uint8 decimals) = gasPayingToken();
+        require(token != Constants.ETHER, "OptimismPortal: only custom gas token");
+
+        // Get the balance of the portal before the transfer.
+        uint256 balanceOf = IERC20(token).balanceOf(address(this));
+
+        // Take ownership of the token. It is assumed that the user has given the portal an approval.
+        Permit2Lib.safeTransferFrom2({ _token: token, _from: msg.sender, _to: address(this), _amount: _mint });
+
+        // Double check that the portal now has the exact amount of token.
+        require(IERC20(token).balanceOf(address(this)) == balanceOf + _mint, "OptimismPortal: transferFrom failed");
+
+        // Normalize to 18 decimals.
+        uint256 mint = Decimals.scale({ _amount: _mint, _decimals: decimals, _target: 18 });
```

Comment:
We want to remove the scaling logic so we can delete `Decimals` library and its tests